pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1656,19 +1619,12 @@ private void sendSegmentRefreshMessage(OfflineSegmentZKMetadata segmentZKMetadat
    *    the segment assignment strategy and replicas.
    *
    * @param segmentMetadata Meta-data for the segment, used to access segmentName and tableName.
-   * @throws JsonParseException
-   * @throws JsonMappingException
-   * @throws JsonProcessingException
-   * @throws JSONException
-   * @throws IOException
    */
-  private void addNewOfflineSegment(final SegmentMetadata segmentMetadata)
-      throws JSONException, IOException {
-    final TableConfig offlineTableConfig =
-        ZKMetadataProvider.getOfflineTableConfig(_propertyStore, segmentMetadata.getTableName());
-
-    final String segmentName = segmentMetadata.getName();
+  private void addNewOfflineSegment(final SegmentMetadata segmentMetadata) {
     final String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(segmentMetadata.getTableName());
+    final String segmentName = segmentMetadata.getName();
+    final TableConfig offlineTableConfig = ZKMetadataProvider.getOfflineTableConfig(_propertyStore, offlineTableName);
+    Preconditions.checkNotNull(offlineTableConfig);
```

Comment:
Should this not be a 400 table not found? Will precondition here end up sending back a 500?

Reply:
We already checked that in PinotSegmentUploadRestletResource.uploadSegment()